http/address_watcher.go

```
@@ -0,0 +1,118 @@
+	"github.com/rsksmart/liquidity-provider/types"
+	log "github.com/sirupsen/logrus"
+	"math/big"
+)
+
+type BTCAddressWatcher struct {
+	btc             connectors.BTCInterface
+	rsk             connectors.RSKInterface
+	lp              providers.LiquidityProvider
+	calledForUser   bool
+	registeredPegIn bool
+	quote           *types.Quote
+}
+
+func NewBTCAddressWatcher(btc connectors.BTCInterface, rsk connectors.RSKInterface, provider providers.LiquidityProvider, q *types.Quote) (*BTCAddressWatcher, error) {
+	watcher := BTCAddressWatcher{
+		btc:             btc,
+		rsk:             rsk,
+		lp:              provider,
+		quote:           q,
+		calledForUser:   false,
+		registeredPegIn: false,
+	}
+	return &watcher, nil
+}
+
+func (w *BTCAddressWatcher) RegisteredPegIn() bool {
+	return w.registeredPegIn
+}
+
+func (w *BTCAddressWatcher) OnNewConfirmation(txHash string, confirmations int64, amount float64) {
+	if !w.calledForUser && confirmations >= int64(w.quote.Confirmations) {
+		_, err := w.performCallForUser()
+		if err != nil {
+			log.Errorf("error calling callForUser. value: %v. error: %v", txHash, err)
+			return
+		}
+		w.calledForUser = true
+	}
+
+	if w.calledForUser && confirmations >= w.rsk.GetRequiredBridgeConfirmations() {
+		_, err := w.performRegisterPegIn(txHash)
+		if err != nil {
+			log.Errorf("error calling registerPegIn. value: %v. error: %v", txHash, err)
+			return
+		}
+		w.registeredPegIn = true
+	}
+}
+
+func (w *BTCAddressWatcher) performCallForUser() (*gethTypes.Transaction, error) {
+	q, err := w.rsk.ParseQuote(w.quote)
+	opt := w.getTxOptions(q.GasLimit.Uint64()+100000, q.Value, q.LiquidityProviderRskAddress)
+	if err != nil {
+		return nil, err
+	}
+	res, err := w.rsk.CallForUser(opt, q)
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+func (w *BTCAddressWatcher) performRegisterPegIn(txHash string) (*gethTypes.Transaction, error) {
+	q, err := w.rsk.ParseQuote(w.quote)
```

This error is not being checked

---

done.